src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -755,6 +755,45 @@ public Geometry GetLocation()
             }
         }
 
+        /// <summary>
+        /// Gets all elements hosted to this host
+        /// </summary>
+        /// <param name="includeOpenings">Include rectangular openings in output</param>
+        /// <param name="includeShadows">Include shadows in output</param>
+        /// <param name="includeEmbeddedWalls">Include embedded walls in output</param>
+        /// <param name="includeSharedEmbeddedInserts">Include shared embedded elements in output</param>
+        /// <returns>Hosted Elements</returns>
+        public Element[] GetHostedElements(bool includeOpenings = false,
+                                           bool includeShadows = false,
+                                           bool includeEmbeddedWalls = false,
+                                           bool includeSharedEmbeddedInserts = false)
+        {
+            if (this.InternalElement == null)
+                throw new Exception(nameof(this.InternalElement));
+
+            HostObject hostObject = this.InternalElement as HostObject;
+            if (hostObject == null)
+                throw new Exception(nameof(hostObject));
+
+            // Document to work with
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+
+            IList<ElementId> inserts = hostObject.FindInserts(includeOpenings,
+                                                              includeShadows,
+                                                              includeEmbeddedWalls,
+                                                              includeSharedEmbeddedInserts);
+
+            // Get all hosted elements from their Id's 
+            // and convert them to DS type
+            var hostedElements = inserts
+                .Select(insert => document
```

Comment:
maybe instead of LINQ and `Select` statements, we should use a simple `for` loop as there's no querying etc and will be faster.

if not, use a LINQ `ForEach` ?